### PR TITLE
Fix #521: allow hyphen in names, prevent empty String

### LINKED ISSUE #521: JSON API should allow hyphens (`-`) in document property names (but should not allow empty String)
Currently only following characters are as follows:

```
[a-zA-Z0-9_]*
```

meaning that ascii letters (both cases), ten-based numbers, and underscore are allowed.
But nothing else.

It seems that there is wide-spread usage of hyphens for Mongoose apps, so we should probably allow that as well: we handle validation and do not rely on Cassandra's column name limitations.

In addition, we should NOT allow empty String as property name as that is not allowed by other databases.

### PR DESCRIPTION
**Pull request by tatu-at-datastax**
**What this PR does**:

Improves property name validation wrt #521 so that:

- Allow hyphen (`-`) in names and name segments
- Empty names no longer allowed (shouldn't have been, oversight)

**Which issue(s) this PR fixes**:
Fixes #521

**Checklist**
- [ ] Changes manually tested
- [x] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

**Review by vkarpov15 — APPROVED** [review]
LGTM :+1:

**Comment by tatu-at-datastax** [comment]
@maheshrajamani Could you also review this when you get a chance?

**Review by maheshrajamani — APPROVED** [review]
@tatu-at-datastax Can you add insert and find (filter) IT for this case?

**Comment by tatu-at-datastax** [comment]
> @tatu-at-datastax Can you add insert and find (filter) IT for this case?

Yes that sounds reasonable.

File: src/test/java/io/stargate/sgv2/jsonapi/api/v1/InsertIntegrationTest.java [hunk 1]
```diff
@@ -67,22 +67,47 @@ public void insertDocument() {
           .body("status.insertedIds[0]", is("doc3"))
           .body("data", is(nullValue()))
           .body("errors", is(nullValue()));
+      given()
```
Comment (tatu-at-datastax) [inline src/test/java/io/stargate/sgv2/jsonapi/api/v1/InsertIntegrationTest.java:70]: Existing not change, just inlined JSON payloads for bit more compact code.

**Comment by tatu-at-datastax** [comment]
@maheshrajamani Added insert and find test cases, will merge once CI passes.

**Pull request merged**